### ISSUE TITLE
The plugin is not working with Elgg 2.3.4

### ISSUE THREAD

**Issue by rohit1290** [iionly/elggx_fivestar#6]
I have this plugin enabled for group rating. Earlier it was working fine but now it seems like not working.

Here's the configuration screenshot:
![settings](https://user-images.githubusercontent.com/2772844/31790614-b1fd7162-b533-11e7-9369-a4bedb3e5597.PNG)

Value that I have entered: 
`elggx_fivestar_view=group/default, tag=div, attribute=class, attribute_value=elgg-subtext, after_html=<br>`

Here's the group screenshot:
![group](https://user-images.githubusercontent.com/2772844/31790668-dc0c6a62-b533-11e7-8a36-55e6416969c2.PNG)

The rating starts used to appear below the name but now its not showing.

**Comment by iionly** [comment]
Elggx Fivestar uses a different view for group ratings on Elgg 2 than it did for Elgg 1. The new view is
```
elggx_fivestar_view=groups/profile/summary, tag=div, attribute=class, attribute_value=groups-stats, before_html=<br>
```
The rating will appear on the group profile page below the group's image.

Use the latest version of the Elggx Fivestar plugin (currently 2.3.2). If you still have problems with the rating widgets, you might need to restore the default rating views (in this case you might need to re-add any custom rating views you added on your own).

But as it is the Fivestar plugin works as intended on Elgg 2.3.4 with regards to the group ratings.

**Comment by rohit1290** [comment]
Hi iionly, I am using the latest version of the Elggx Fivestar plugin (2.3.2) with Elgg 2.3.4
I updated the "groups/profile/summary" tag but still I can't see the plugin showing any stars in my group profile.

Here the setting page now:
![settings](https://user-images.githubusercontent.com/2772844/31834736-a2bf739e-b5ec-11e7-877a-c745172c65f8.PNG)

Here's the group profile page
![profile](https://user-images.githubusercontent.com/2772844/31834745-af939618-b5ec-11e7-9f67-e7d0148cbaf2.PNG)

**Comment by iionly** [comment]
Possibly, your theme causes the rating widget to not appear. Have you tried with the theme plugin disabled to see if it works then?

**Comment by rohit1290** [comment]
Yes, already tried that. Disabling the theme did not solve the problem.
Also, earlier it was working just fine with this theme.

**Comment by iionly** [comment]
Maybe something has gone wrong on upgrading the Fivestar plugin then. As I mentioned, it works perfectly fine for me on Elgg 2.3.4.

Try: disabling the Fivestar plugin, remove the fivestar plugin folder, then copy the latest version of the plugin again in the mod folder and then re-activate the Fivestar plugin. Next you should also reset the rating views to the default view on the Fivestar settings page. If it still doesn't work then, disable the 3rd party plugins on your site temporarily to see if any of them causes the problem. I'm afraid I can't give any other hints at the moment if I can't reproduce the issue.

**Comment by rohit1290** [comment]
I did a complete new setup of elgg in my local system. I downloaded the latest version of Elgg and Five Star plugin. I configured the plugin from scratch and saw the problem still exist, I can't see the five star widget in the group page.

Here's the screenshot
![capture](https://user-images.githubusercontent.com/2772844/31898282-68e10cae-b836-11e7-806b-c0b6f917913b.PNG)

To reproduce the error, you can do a fresh elgg setup and check if the plugin is working or not.

**Comment by iionly** [comment]
Are you really sure that you use the right version of Elggx Fivestar (version 2.3.2)? And are you testing with the default rating views?

I just tried with a fresh install of Elgg 2.3.4 and it gave the same result as with my other Elgg 2.3.4 installation. It just works:

![screenshot_20171023_200109](https://user-images.githubusercontent.com/1773681/31905116-65628aa2-b82d-11e7-85f7-ee267301b8cd.png)

**Comment by rohit1290** [comment]
Yes, I am using Elgg 2.3.4

![1](https://user-images.githubusercontent.com/2772844/31938749-0c042458-b8d6-11e7-9752-b8228078ae50.PNG)

And Five Star 2.3.2

![2](https://user-images.githubusercontent.com/2772844/31938771-192a4bbc-b8d6-11e7-8987-f1172a8687f5.PNG)

With the default view

![3](https://user-images.githubusercontent.com/2772844/31938872-5503bcc2-b8d6-11e7-94c5-b4de03878074.PNG)

Can you please zip and send me your local copy of elggx fivestar plugin at rohit_1290@yahoo.com

**Comment by rohit1290** [comment]
The plugin is working fine. **the only problem is that the stars are not appearing in the view**.
For now a workaround that i am using is that, I created a copy of the group summary file in a new plugin where I am calling the elgg_view "elggx_fivestar/voting" manually.

**Comment by iionly** [comment]
The version I have is absolutely the same as available at https://elgg.org/plugins/843333/releases/2.3.2 or here on github as 2.3.2 release. I don't know why it should fail to show the stars for you.

Do you have any caching tools installed on your server (proxy, optcache) or are you using any offserver cache or CDN? In this case there might be some outdated view still in any of these caches. Other than a caching issue I don't know why it should fail to work for you. But even the caching shouldn't be an issue on a new installation from scratch without any other 3rd party plugins installed or any code modified.

**Comment by rohit1290** [comment]
I finally found the problem. The plugin is working fine for PHP version 5, however the stars are not appearing in PHP version 7.

![problem](https://user-images.githubusercontent.com/2772844/31987469-8d9a0d6c-b989-11e7-9e2b-696b637fd2fc.PNG)

My Elgg is running on PHP 7.0.16 and that is why the stars are not appearing but if I run the plugin in PHP 5.6.30 the plugin is working absolutely fine.

I believe you are running your elgg in PHP 5 and hence it's working fine for you.

**Comment by iionly** [comment]
I guess I haven't fully understood that you don't see the stars anywhere. I was assuming that it only failed to work for the group rating but worked fine everywhere else.

PHP 7 as the reason for the plugin not fully working could indeed be the case. Unfortunately, I can switch to PHP 7 yet on my development server (only way I could test) because I still need PHP 5.6 due to backward compatibility with older Elgg versions. Maybe I can get a PHP 7 installation working for testing on my old notebook but I can't say for sure that I will get this ever working again as it's really not easy anymore to get a recent Linux distribution working on a 32bit machine (only 14 years old...lol).

Do you see any errors in the log? It could help me to know if there's any single issue with PHP 7 only. My guess is that it might be the dom parser script used as it's not really a recent version (but no newer version of this script exists). So, it might be necessary to do some major rework here...

**Comment by rohit1290** [comment]
> I guess I haven't fully understood that you don't see the stars anywhere. I was assuming that it only failed to work for the group rating but worked fine everywhere else.

No, the star are not appearing anywhere, I am using it for group rating hence my main concern was group.

> Do you see any errors in the log?

I have tried every possible way like enabling the PHP log using error_reporting, developer plugin, self testing by tweaking some code but as of now I am not getting any error in the log that can help us point to the issue.

> Unfortunately, I can switch to PHP 7 yet on my development server (only way I could test) because I still need PHP 5.6 due to backward compatibility with older Elgg versions.

As of now I am working with the workaround. But it would be great if this issue can be solved.

> Maybe I can get a PHP 7 installation working for testing on my old notebook but I can't say for sure that I will get this ever working again as it's really not easy anymore to get a recent Linux distribution working on a 32bit machine (only 14 years old...lol).

I can give you ftp access to my test server but I think developing on server end will be a challenge.

**Comment by iionly** [comment]
It looks like it's easy to fix. I got my notebook working at least so far that I could test it out. I can reproduce the issue.

Fix seems simple. Just change line 173 of start.php from
```
if ($element->$options['attribute'] == $options['attribute_value']) {
```
into
```
if ($element->{$options['attribute']} == $options['attribute_value']) {
```
This seems the only change necessary for the stars to appear again on php7. Could you test, please? If it works for you, too, I can make a new release.

**Comment by rohit1290** [comment]
Perfect!
This resolved the php7 problem. The stars are appearing again. 
I have tested it for bookmarks, blog, files, discussion, album and groups.

**Comment by iionly** [comment]
Fixed in version 2.3.3.
